lib/ar2dto/has_dto.rb

```diff
@@ -22,17 +22,52 @@ def has_dto
       end
 
       # @api public
-      def to_dto
-        all.map(&:to_dto)
+      def to_dto(options = {})
+        all.map { |record| record.to_dto(options) }
       end
     end
 
     # Wrap the following methods in a module so we can include them only in the
     # ActiveRecord models that declare `has_dto`.
     module InstanceMethods
       # @api public
-      def to_dto
-        "#{self.class.name}DTO".constantize.new(attributes)
+      def to_dto(options = {})
+        options = apply_global_configs(options)
+
+        hash = serializable_hash(options&.except(:include))
+        hash = add_associations(hash, options&.dig(:include))
+
+        "#{self.class.name}DTO".constantize.new(hash)
+      end
+
+      private
+
+      def apply_global_configs(options)
+        options[:except] = AR2DTO::Config.instance.except + Array(options[:except])
+        options
+      end
+
+      def add_associations(hash, includes)
+        includes_to_dto(includes) do |association, records, opts|
+          hash[association.to_s] = if records.respond_to?(:to_ary)
+                                     records.to_ary.map { |a| a.to_dto(opts) }
+                                   else
+                                     records.to_dto(opts)
+                                   end
+        end
+        hash
+      end
+
+      def includes_to_dto(includes)
```

Comment:
🪄 🎩

Reply:
I stole this code from the Rails codebase 🤣